### PR TITLE
Remove unused New User Onboarding Modal

### LINKED ISSUE #14128: Remove `NewUserOnboardingModal` and related files
During some work on unit tests, I discovered an unknown code in `NewUserOnboardingModal.unit.spec.js` and `StepIndicator.unit.spec.js`. @mazameli told me that modal was not in use for quite some time (around 2 years).

We should remove it and any related code around it.

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Closes #14128 
- Removes long time unused set of components related to the new user onboarding modal

### Additional notes and questions
- `HomePage.jsx` is actually an `/activity` page? Is that also a part of the tech debt that should be fixed maybe?
- What should I do with the translations, @mazameli ? Are they going to be automatically removed from the POeditor?
- Please check for any places that still reference this old code which I might have missed.

### CODE REVIEW

File: frontend/src/metabase/home/containers/HomepageApp.jsx [hunk 1]
```diff
@@ -52,27 +48,12 @@ export default class HomepageApp extends Component {
     super(props);
     this.state = {
       greeting: Greeting.sayHello(props.user && props.user.first_name),
```
Comment (nemanjaglumac) [inline frontend/src/metabase/home/containers/HomepageApp.jsx:50]: I don't think this is used anywhere on the page.

Comment (nemanjaglumac) [reply frontend/src/metabase/home/containers/HomepageApp.jsx]: Removed it in https://github.com/metabase/metabase/pull/14129/commits/8f07cd0fe8738fbd9ac3d3309ae9329f31dda289. If that wasn't the right move, I can easily revert it.

File: frontend/src/metabase/home/containers/HomepageApp.jsx [hunk 2]
```diff
@@ -4,14 +4,11 @@ import { connect } from "react-redux";
 import { t } from "ttag";
 import { push } from "react-router-redux";
 
-import Greeting from "metabase/lib/greeting";
```
Comment (mazameli) [inline frontend/src/metabase/home/containers/HomepageApp.jsx:7]: This shouldn't be removed. It's what powers the "MetaBot" greetings in the top-left of the homepage:

image

File: frontend/src/metabase/home/containers/HomepageApp.jsx [hunk 3]
```diff
@@ -50,29 +45,11 @@ export default class HomepageApp extends Component {
 
   constructor(props) {
     super(props);
-    this.state = {
-      greeting: Greeting.sayHello(props.user && props.user.first_name),
```
Comment (mazameli) [inline frontend/src/metabase/home/containers/HomepageApp.jsx:54]: I'm guessing we don't want to remove this either — right @kdoh?

**Review by mazameli — CHANGES_REQUESTED** [review]
Generally looks good to me except for removing the homepage greeting. I'd also feel more comfortable with this PR if @kdoh could take a quick look.

File: frontend/src/metabase/home/containers/HomepageApp.jsx (hunk 2 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:7]: I still have it in `localhost` on the homepage, after having removed this.
image

File: frontend/src/metabase/home/containers/HomepageApp.jsx (hunk 3 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:54]: If I'm not mistaken, there is a naming confusion here.
`HomePage.jsx` is actually `/activity` page, while
`Overworld.jsx` is a home page.

There is a `Greeting()` function in Subhead at `frontend/src/metabase/containers/Overworld.jsx:95`.

File: frontend/src/metabase/home/containers/HomepageApp.jsx [hunk 4]
```diff
@@ -23,7 +20,6 @@ const mapStateToProps = (state, props) => ({
   activity: getActivity(state),
   recentViews: getRecentViews(state),
   user: getUser(state),
```
Comment (paulrosenzweig) [inline frontend/src/metabase/home/containers/HomepageApp.jsx:22]: I think this and the proptype can be deleted too.

**Review by paulrosenzweig — APPROVED** [review]
LGTM

File: frontend/src/metabase/home/containers/HomepageApp.jsx (hunk 3 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:54]: Yeah `greeting` is never accessed in the component state here.

File: frontend/src/metabase/home/containers/HomepageApp.jsx (hunk 4 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:22]: I tried, but it seems it is used by (passed down to) `Activity.jsx` component. I you remove it, there is a console error:
```
warning.js:33 Warning: Failed prop type: The prop `user` is marked as required in `Activity`, but its value is `undefined`.
    in Activity (created by HomepageApp)
    in HomepageApp (created by HomepageApp)
    in HomepageApp (created by RouterContext)
    in Unknown (created by UserIsAuthenticated(Component))
    in UserIsAuthenticated(Component) (created by Connect(UserIsAuthenticated(Component)))
    in Connect(UserIsAuthenticated(Component)) (created by MetabaseIsSetup(Connect(UserIsAuthenticated(Component))))
    in MetabaseIsSetup(Connect(UserIsAuthenticated(Component))) (created by Connect(MetabaseIsSetup(Connect(UserIsAuthenticated(Component)))))
    in Connect(MetabaseIsSetup(Connect(UserIsAuthenticated(Component)))) (created by RouterContext)
    in div (created by App)
    in ScrollToTop (created by ScrollToTop)
    in ScrollToTop (created by App)
    in App (created by App)
    in App (created by Title[App])
    in Title[App] (created by RouterContext)
    in RouterContext (created by Router)
    in Router
    in ThemeProvider
    in DragDropContextProvider
    in Provider
    ```

Comment (paulrosenzweig) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:22]: Oops! I missed that.

File: frontend/src/metabase/home/containers/HomepageApp.jsx (hunk 3 above)
Comment (nemanjaglumac) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:54]: @mazameli check this out :)

image

**Review by mazameli — APPROVED** [review]

File: frontend/src/metabase/home/containers/HomepageApp.jsx (hunk 3 above)
Comment (mazameli) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:54]: My bad! Yeah, the Activity page used to be the homepage years ago. :)

Comment (nemanjaglumac) [reply frontend/src/metabase/home/containers/HomepageApp.jsx:54]: I'm learning some history here :)

**Pull request merged**